src/trovares_connector/neo4j_connector/query_translator.py

```diff
@@ -0,0 +1,366 @@
+            if frame_name in self._edge_name_mapping:
+                left_node_labels = left_nodes['labels']
+                if len(left_node_labels) == 0:
+                    source = None
+                elif len(left_node_labels) == 1:
+                    source = list(left_node_labels.keys())[0]
+                else:
+                    raise ValueError(f"Support for multi-label left nodes not yet support: {left_nodes}")
+                right_node_labels = right_nodes['labels']
+                if len(right_node_labels) == 0:
+                    target = None
+                elif len(right_node_labels) == 1:
+                    target = list(right_node_labels.keys())[0]
+                else:
+                    raise ValueError(f"Support for multi-label nodes not yet support: {right_nodes}")
+
+                swap_source_target = relationship['left_arrow'] and not relationship['right_arrow']
+                if swap_source_target:
+                    temp = source
+                    source = target
+                    target = temp
+                
+                if self._verbose:
+                    print(f"----> Source: {source}")
+                    print(f"----> Target: {target}")
+
+                if source is not None and target is not None:
+                    mapping = self._edge_name_mapping[frame_name]
+                    if self._verbose:
+                        print(f"Frame to adapt: (:{source})-[:{frame_name}]->(:{target})")
+                        print(f"  Frame Name {frame_name} needs adapting, {len(mapping)} mapped frames")
+                    for variant in mapping:
+                        if variant['source'] == source and variant['target'] == target:
+                            if self._verbose:
+                                print(f"Use this variant: {variant}")
+                            reltype_loc = relationship['rel_types'][frame_name].location
+                            if reltype_loc['offset'] in self._rewrites:
+                                raise ValueError(f"Multiple rewrites at offset: {reltype_loc['offset']}")
+                            self._rewrites[reltype_loc['offset']] = {
+                                'from':reltype_loc, 
+                                'to':variant['mapped_name']}
+
+    def _rewrite_query(self, query:str) -> str:
+        """Produce a re-written query from the self._rewrites information."""
+        if self._verbose:
+            print(f"Rewrites:")
+            pprint(self._rewrites)
+        locations = list(self._rewrites.keys())
+        locations.sort(reverse = True)
+        for offset in locations:
+            rewrite = self._rewrites[offset]
+            length = rewrite['from']['length']
+            new_text = rewrite['to']
+            query = f"{query[:offset]}{new_text}{query[offset+length:]}"
+
+        return query
+
+
+class QueryElement(object):
+    """
+    Utility object to store an element of a query.
+    
```

Comment:
There are extra spaces here.

Reply:
Was this the trailing blank on line 186?  Or are you saying the blank line between the brief description and the detailed description?